### ISSUE TITLE
Support SWAP TABLES

### ISSUE THREAD

**Issue by rjnn** [MaterializeInc/materialize#5680]
Support `ALTER TABLE foo SWAP WITH bar;`

The impetus here is that folks will build up a new view and want to swap out the old materialized view with the new one, while ensuring that no SELECTs get `VIEW DOES NOT EXIST` in the interim.

Not sure what should happen with sinks, but presumably something good can be made to happen.

Inspiration: Snowflake. dbt+snowflake has a common pattern where people build "pipelines" and iterate on a new one while the old one is still up, and then eventually do the indiana jones swap. See https://docs.snowflake.com/en/sql-reference/sql/alter-table.html

## See also
* https://github.com/MaterializeInc/materialize/issues/22250

**Comment by benesch** [comment]
I think the actual request here might be transactional DDL (#3616)?

```sql
BEGIN;
DROP VIEW foo_view;
ALTER RENAME foo_view_new TO foo_view;
COMMIT;
```

`ALTER ... SWAP` is cute but breaks down if you want to swap more than one view atomically.

**Comment by rjnn** [comment]
I think this is a simpler first case than transactional DDL. I wanted to note that Snowflake considers this an important enough pattern to special case out (and seen some internet love for this feature) - and it fits my intuition that folks want to upgrade views one by one.

**Comment by benesch** [comment]
Hmm, ok.

**Comment by morsapaes** [comment]
> and it fits my intuition that folks want to upgrade views one by one.

I agree that the final workflow needs to give users more fine-grained control over what is re-deployed than we can provide with the first blue/green implementation we landed on (see #19748). Tentatively closing this issue since it's old-ish, and we'll need to revisit schema evolution soon in light of fresher issues like #15897 and #26350; more granular object deployment should be discussed then, too.
